core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy

```diff
@@ -300,66 +300,53 @@ class GemVersion implements Comparable<GemVersion> {
      * @since 2.0
      */
     GemVersion intersect(GemVersion other) {
-        Boundary newLowBoundary
-        String newLow
-        switch (compare(low, other.low)) {
-            case -1:
-                newLow = other.low
-                newLowBoundary = other.lowBoundary
-                break
-            case 0:
-                newLowBoundary = (lowBoundary == EXCLUSIVE || other.lowBoundary == EXCLUSIVE) ? EXCLUSIVE : INCLUSIVE
-                newLow = low
-                break
-            case 1:
-                newLow = low
-                newLowBoundary = lowBoundary
+        Tuple2<String, Boundary> newLowVersionSpec = intersect(low, lowBoundary, other.low, other.lowBoundary, true)
```

Comment:
We should probably introduce a data class to encapsulate the version and the boundary:

```groovy
class GemVersion {
    final GemVersionConstraint low
    final GemVersionConstraint high
}

class GemVersionConstraint {
  final String version
  final String boundary
}
```

But I don't want to introduce too much changes in a bug-fix pull request.

Reply:
Feel free to do so in a subsequent pR.